### PR TITLE
Add the missing store creation entry in the release notes

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds the missing **[Internal]** entry in the release notes for the store creation flow.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 11.5
 -----
+- [**] [Internal] Store creation flow was improved with native implementation. It is available from the login prologue (`Get Started` CTA), login email error screen, and store picker (`Add a store` CTA from the empty stores screen or at the bottom of the store list). [Example testing steps in https://github.com/woocommerce/woocommerce-android/pull/7947]
```
Comment (pmusolino) [inline RELEASE-NOTES.txt]: I'm wrong, or the Store Creation flow was released under 11.4?

Comment (JorgeMucientes) [reply RELEASE-NOTES.txt]: Yes this was the [last PR](https://github.com/woocommerce/woocommerce-android/pull/7947) to be merged for the native flow and was merged in `11.4`. I also checked and we set the feature flag `NATIVE_STORE_CREATION_FLOW -> true` in 11.4. 

So we should update that release note.

Comment (JorgeMucientes) [reply RELEASE-NOTES.txt]: Done

**Review by JorgeMucientes — APPROVED** [review]

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (0nko) [reply RELEASE-NOTES.txt]: Oops, thanks for the fix, @JorgeMucientes!

**Pull request merged**